src/test/scala/com/databricks/labs/overwatch/pipeline/InitializeTest.scala

```diff
@@ -0,0 +1,224 @@
+      val conf = new Config
+      conf.setCloudProvider("aws")
+      val init = new Initializer(conf)
+      val quickBuildAuditLogConfig = PrivateMethod[AuditLogConfig]('validateAuditLogConfigs)
+
+      assertThrows[BadConfigException](init invokePrivate quickBuildAuditLogConfig(configInput))
+    }
+
+    it("validateAuditLogConfigs function validate audit log format in the config ") {
+      import com.databricks.labs.overwatch.utils.AuditLogConfig
+
+      val configInput = AuditLogConfig(Some("path/to/audit/log"), "text", None)
+      val conf = new Config
+      conf.setCloudProvider("aws")
+      val init = new Initializer(conf)
+      val quickBuildAuditLogConfig = PrivateMethod[AuditLogConfig]('validateAuditLogConfigs)
+
+      assertThrows[BadConfigException](init invokePrivate quickBuildAuditLogConfig(configInput))
+    }
+  }
+
+
+  describe("Tests for validateAndRegisterArgs function") {
+    it("validateAndRegisterArgs function should validate and register variables on the conf") {
+      val incomplete = "{\"auditLogConfig\":{\"rawAuditPath\":\"/mnt/logs/test/audit_delivery\"}," +
+        "\"tokenSecret\":{\"scope\":\"overwatch\",\"key\":\"1212\"},\"dataTarget\":{\"databaseName\":\"overwatch_etl_test\"," +
+        "\"databaseLocation\":\"dbfs:/mnt/overwatch_global/consume/overwatch_etl.db\",\"consumerDatabaseName\":\"overwatch\"" +
+        ",\"consumerDatabaseLocation\":\"dbfs:/mnt/overwatch_global/consume/overwatch.db\"}," +
+        "\"badRecordsPath\":\"/mnt/overwatch/workspace-1029/sparkEventsBadrecords\",\"overwatchScope\":[\"all\"]," +
+        "\"maxDaysToLoad\":60,\"databricksContractPrices\":{\"interactiveDBUCostUSD\":0.56,\"automatedDBUCostUSD\":0.26}," +
+        "\"primordialDateString\":\"2021-01-16\"}"
+
+      val conf = new Config
+      conf.setOrganizationId("dummyorgifid")
+      val init = new Initializer(conf)
+      val validateAndRegisterArgs = PrivateMethod[Initializer]('validateAndRegisterArgs)
+      init invokePrivate validateAndRegisterArgs(incomplete)
+
+      assert(conf.workspaceName == "dummyorgifid")
+      assert(conf.databaseName == "overwatch_etl_test")
+    }
+
+  }
+
+  describe("Tests for dataTargetIsValid function") {
+    import com.databricks.labs.overwatch.utils.DataTarget
+    it("dataTargetIsValid function should throw exception when the current db location is different than the one present already") {
+      val dataTarget = DataTarget(Some("overwatch_etl"),Some("/path/to/database"), Some("/path/prefix"), Some("overwatch"), Some("/path/to/consumer_database"))
+      spark.sql("create database if not exists overwatch_etl")
+
+      val conf = new Config
+      val init = new Initializer(conf)
+      val dataTargetIsValid = PrivateMethod[Boolean]('dataTargetIsValid)
+      assertThrows[BadConfigException](init invokePrivate dataTargetIsValid(dataTarget))
+    }
+
+    it("dataTargetIsValid function should throw exception when the current db is not created from overwatch") {
```

Comment:
more tests:

- use initializeDatabase to create a database that is owned by overwatch and validate that it works as expected.
- additional tests around initDatabase and validate DataTarget for schema tests
- I don't see any tests for the metadata checks
- what if consumerDB is left blank in config -- does it work as expected (i.e. etl and consumer dbs are the same)?

I think there are likely more scenarios than this even...I want to make sure and have good coverage on `initDatabase` and `validateDataTarget`. Please feel free to ask some questions but I think if you review the code a little deeper you can find quite a few more cases.